module/zfs/spa.c

```diff
@@ -4235,6 +4420,8 @@ spa_tryimport(nvlist_t *tryconfig)
 		    poolname) == 0);
 		VERIFY(nvlist_add_uint64(config, ZPOOL_CONFIG_POOL_STATE,
 		    state) == 0);
+		VERIFY(nvlist_add_uint64(config, ZPOOL_CONFIG_IMPORT_TXG,
+		    spa->spa_uberblock.ub_txg) == 0);
```

Comment:
@ofaaland in order to address the issue @nedbass found we should only add `ZPOOL_CONFIG_IMPORT_TXG` to the config when the activity check passes.  The existing code adds it regardless which isn't what was intended.  I'd suggest moving this code to the success path of the `spa_activity_check()` function.